frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -266,6 +393,36 @@ describe('StudioCodeListEditor', () => {
   });
 });
 
-function renderCodeListEditor(props: Partial<StudioCodeListEditorProps> = {}) {
+function renderCodeListEditor(props: Partial<StudioCodeListEditorProps> = {}): RenderResult {
   return render(<StudioCodeListEditor {...defaultProps} {...props} />);
 }
+
+async function switchToSearchMode(
+  user: UserEvent,
+  textPropertyCoords: TextPropertyCoords,
+): Promise<void> {
+  const name = texts.textResourceTexts(...textPropertyCoords).search;
+  const searchModeButton = screen.getByRole('radio', { name });
+  await user.click(searchModeButton);
+}
+
+type TextPropertyCoords = [number, CodeListItemTextProperty];
+
+function getTextResourcePicker(textPropertyCoords: TextPropertyCoords): HTMLElement {
+  const { textResourcePickerLabel } = texts.textResourceTexts(...textPropertyCoords);
+  return screen.getByRole('combobox', { name: textResourcePickerLabel });
+}
+
+function getTextResourceValueInput(textPropertyCoords: TextPropertyCoords): HTMLElement {
+  const { valueLabel } = texts.textResourceTexts(...textPropertyCoords);
```

Comment:
Any reason why these two name-consts are extracted from the returned object and not retuned directly as in the `switchToSearchMode` and `getTextResourceOption`?

No issue, but would be nice to keep consistency 🤗

Reply:
Not sure if I'm hanging along. What do you mean by name-consts?

Reply:
In the switch function and in the `getTextResourceOption` test, the `name` const can be used directly in the `getByRole`. But in these two tests the name value is specified with the value in the returned object. Why is it not consistent like this:
`const name = texts.textResourceTexts(...textPropertyCoords).valueLabel;`

or like this in the search function:

```
const { search } = texts.textResourceTexts(...textPropertyCoords);
  const searchModeButton = screen.getByRole('radio', { name: search });
```

But again, no biggie 😊

Reply:
I understand. As long as we keep the lines within our max length, I'm happy. I've changed this now.